src/d3a/models/strategy/external_strategies/__init__.py

```diff
@@ -217,16 +218,18 @@ def next_market(self):
     def _get_market_from_command_argument(self, arguments: Dict):
         if "timeslot" not in arguments:
             return self.next_market
-        return self._get_market_from_timeslot(arguments["timeslot"])
+        timeslot = str_to_pendulum_datetime(arguments["timeslot"])
+        return self._get_market_from_timeslot(timeslot)
 
     def _get_market_from_timeslot(self, timeslot: DateTime):
         market = self.market_area.get_market(timeslot)
         if market:
             return market
         market = self.market_area.get_settlement_market(timeslot)
-        if market:
-            return market
-        return None
+        if not market:
+            raise Exception(f"Timeslot {timeslot} is not currently in the spot, future or "
```

Comment:
I would suggest to use a specific exception, maybe one of the ones we implemented here: https://github.com/gridsingularity/d3a/blob/d6926537004f1ef411e5988ee7e8c2025957bed0/src/d3a/d3a_core/exceptions.py

Reply:
@fievelk Lets discuss this topic again next time we are in the office ok? I am still not convinced by this rule, I know it is in PEP8, but isn't the message of the exception already specific enough? Why creating a special Exception for this specific case? 
Again, no need to answer now. You can pitch me this topic in person. :-D

Reply:
Sure! I'll practice my public-speaking skills 😄

Reply:
There is one advantage in subclassing and using specific exception types: grouping by type, and catching exceptions of a specific type and handling them differently than other exceptions. Unfortunately here the exception does not really help much because the exception handler catches all exception types and propagates them to the response of the API command. That's why I am also not really sold that a specific exception type is needed here. Having said all of the above, looking forward to discussing this in person, the discussion will definitely be interesting for me!